src/hotspot/share/services/threadService.cpp

```diff
@@ -217,6 +221,7 @@ void ThreadService::current_thread_exiting(JavaThread* jt, bool daemon) {
   assert(!jt->is_terminated() && jt->is_exiting(), "must be exiting");
 
   decrement_thread_counts(jt, daemon);
+  ThreadService::incr_exited_allocated_bytes(jt->cooked_allocated_bytes());
```

Comment:
Again this is too soon. This should be deferred until after the thread has removed itself from the threads-list.

Reply:
Done.